data-products/src/main/scala/org/sunbird/analytics/job/report/CollectionSummaryJob.scala

```diff
@@ -0,0 +1,183 @@
+    val courseInfo = CourseUtils.getCourseInfo(courseIds, 500).toDF("framework", "identifier", "name", "channel", "batches", "organisation")
+    JobLogger.log(s"Total courseInfo records ${courseInfo.count()}", None, INFO)
+    val filteredBatches = processBatches.join(courseInfo, processBatches.col("courseid") === courseInfo.col("identifier"), "inner")
+      .select(processBatches.col("*"), courseInfo.col("identifier"), courseInfo.col("channel"), courseInfo.col("name"), courseInfo.col("organisation"))
+      .withColumn("collectionName", col("name"))
+      .withColumn("publishedBy", concat_ws(", ", col("organisation")))
+      .withColumn("isCertified", when(col("certificates").isNotNull && size(col("certificates").cast("array<map<string, string>>")) > 0, "Y")
+        .when(col("issued_certificates").isNotNull && size(col("issued_certificates").cast("array<map<string, string>>")) > 0, "Y").otherwise("N"))
+    computeValues(filteredBatches.drop("issued_certificates", "certificates", "userchannel", "identifier", "channel", "name", "organisation").persist(StorageLevel.MEMORY_ONLY))
+  }
+
+  def computeValues(transformedDF: DataFrame): DataFrame = {
+    // Compute completionCount and enrolCount for state
+    val statePartitionDF = transformedDF.groupBy("batchid", "courseid", "userstate").agg(
+      count(when(col("status") === 2, 1)).as("completionUserCountByState"),
+      count(when(col("isCertified") === "Y", 1)).as("certificateIssueCount"),
+      count(col("userid")).as("enrolledUsersCountByState")
+    )
+    statePartitionDF.join(transformedDF.drop("isCertified", "status", "userstate").dropDuplicates("courseid", "batchid"), Seq("courseid", "batchid"), "inner")
+      .withColumn("batchid", concat(lit("batch-"), col("batchid")))
+      .withColumn("hasCertified", when(col("certificateIssueCount") > 0, "Y").otherwise("N"))
+      .select("publishedBy", "batchid", "courseid", "collectionName", "startdate", "enddate", "hasCertified", "userstate", "enrolledUsersCountByState", "completionUserCountByState", "certificateIssueCount")
+  }
+
+  def saveToBlob(reportData: DataFrame): Unit = {
+    val container = AppConf.getConfig("cloud.container.reports")
+    val objectKey = AppConf.getConfig("course.metrics.cloud.objectKey")
+    val storageConfig = getStorageConfig(container, objectKey)
+    JobLogger.log(s"Uploading reports to blob storage", None, INFO)
+    val reportPath = "collection-summary-reports/"
+    val fields = reportData.schema.fieldNames
+    val colNames = for (e <- fields) yield columnMapping.getOrElse(e, e)
+    val dynamicColumns = fields.toList.filter(e => !columnMapping.keySet.contains(e))
+    val columnWithOrder = (columnsOrder ::: dynamicColumns).distinct
+    reportData.toDF(colNames: _*).select(columnWithOrder.head, columnWithOrder.tail: _*)
+      .saveToBlobStore(storageConfig, "csv", s"${reportPath}summary-report-${getDate}", Option(Map("header" -> "true")), None)
```

Comment:
File name should always be constant. Create a file with <report-id>_<date>.csv and <report-id>_latest.csv